receiver/gitproviderreceiver/internal/scraper/githubscraper/github_scraper.go

```diff
@@ -65,6 +63,74 @@ func (ghs *githubScraper) scrape(_ context.Context) (pmetric.Metrics, error) {
 	currentDate := time.Now().Day()
 	ghs.logger.Sugar().Debugf("current date: %v", currentDate)
 
-	ghs.logger.Sugar().Debug("github scraper has started")
+	// Enable the ability to override the endpoint for self-hosted github instances
+	// GitHub Free URL : https://api.github.com/graphql
+	// https://docs.github.com/en/graphql/guides/forming-calls-with-graphql#the-graphql-endpoint
+	graphCURL := "https://api.github.com/graphql"
+
+	if ghs.cfg.HTTPClientSettings.Endpoint != "" {
+		var err error
+
+		// GitHub Enterprise (ghe) URL : http(s)://HOSTNAME/api/graphql
+		// https://docs.github.com/en/enterprise-server@3.8/graphql/guides/forming-calls-with-graphql#the-graphql-endpoint
+		graphCURL, err = url.JoinPath(ghs.cfg.HTTPClientSettings.Endpoint, "api/graphql")
+		if err != nil {
+			ghs.logger.Sugar().Errorf("error: %v", err)
+		}
+	}
+	ghs.logger.Sugar().Debugf("GitHub GraphQL endpoint URL set to: %v", graphCURL)
+
+	genClient := graphql.NewClient(graphCURL, ghs.client)
+
+	// Do some basic validation to ensure the values provided actually exist in github
+	// prior to making queries against that org or user value
+	exists, ownertype, err := ghs.checkOwnerExists(ctx, genClient, ghs.cfg.GitHubOrg)
+	if err != nil {
+		ghs.logger.Sugar().Errorf("error checking if owner exists", zap.Error(err))
+	}
+
+	typeValid, err := checkOwnerTypeValid(ownertype)
+	if err != nil {
+		ghs.logger.Sugar().Errorf("error checking if owner type is valid", zap.Error(err))
+	}
+
+	if !exists || !typeValid {
+		ghs.logger.Sugar().Error("error logging in and getting data from github")
+		return ghs.mb.Emit(), err
+	}
+
+	// Generate the search query based on the type, org/user name, and the search_query
+	// value if provided
+	sq := genDefaultSearchQuery(ownertype, ghs.cfg.GitHubOrg)
+
+	if ghs.cfg.SearchQuery != "" {
+		sq = ghs.cfg.SearchQuery
+		ghs.logger.Sugar().Debugf("using search query where query is: %v", ghs.cfg.SearchQuery)
+	}
+
+	// Get the repository data based on the search query retrieving a slice of branches
+	// and the recording the total count of repositories
+	repos, count, err := ghs.getRepos(ctx, genClient, sq)
+	if err != nil {
+		ghs.logger.Sugar().Errorf("error getting repo data", zap.Error(err))
+		return ghs.mb.Emit(), err
+	}
+
+	ghs.mb.RecordGitRepositoryCountDataPoint(now, int64(count))
+
+	// Get the branch count (future branch data) for each repo and record the given metrics
+	for _, repo := range repos {
+		name := repo.(*SearchNodeRepository).Name
+		trunk := repo.(*SearchNodeRepository).DefaultBranchRef.Name
+
+		count, err := ghs.getBranches(ctx, genClient, name, ghs.cfg.GitHubOrg, trunk)
+		if err != nil {
+			ghs.logger.Sugar().Errorf("error getting branch count", zap.Error(err))
+			return ghs.mb.Emit(), err
```

Comment:
maybe could continue here instead of returning early, just a thought.

Reply:
yup, absolutely, good call!

Reply:
ended up just putting these in a go routine & removed the returns, solely choosing to log the error